File: leetcode/easy/find-numbers-with-even-number-of-digits.go
package main

import "fmt"

func findNumbers(nums []int) int {
	result := 0

	for _, num := range nums {
		if num == 100000 {
			result++
			continue
		}

		for _, t := range []int{100, 10000} {
			target := num % t
			if target == num {
				if 10 <= target && target < 100 {
					result++
					break
				} else if 1000 <= target && target < 10000 {
					result++
					break
				}
			}
		}
	}

	return result
}

func main() {
	fmt.Println(findNumbers([]int{12, 345, 2, 6, 7896}))
	fmt.Println(findNumbers([]int{555, 901, 482, 1771}))
}

File: leetcode/easy/find-numbers-with-even-number-of-digits.cpp
#include <iostream>
#include <vector>
using namespace std;

int abs(int target) {
  if (0 < target) {
    return target;
  }
  return -target;
}

int check(int target) {
  int count = 0;

  while (target >= 10) {
    target /= 10;
    count++;
  }

  return count + 1;
}

int findNumbers(vector<int> &nums) {
  int count = 0;

  for (int i = 0; i < nums.size(); i++) {
    if (check(nums[i]) % 2 == 0) {
      count++;
    }
  }

  return count;
}

int main() {
  vector<int> nums{12, 345, 2, 6, 7896};
  int result = findNumbers(nums);

  cout << result;
}